specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2020-06-01-preview/nova.json

```diff
@@ -0,0 +1,1051 @@
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted. Update will proceed asynchronously."
+          },
+          "200": {
+            "description": "The managed Cassandra data center was retrieved successfully.",
+            "schema": {
+              "$ref": "#/definitions/DataCenterInstanceViewResource"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/DefaultErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "clusterNameParameter": {
+      "name": "clusterName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "description": "Managed Cassandra cluster name.",
+      "minLength": 1,
+      "maxLength": 50,
+      "pattern": "^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*"
+    },
+    "dataCenterNameParameter": {
+      "name": "dataCenterName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "description": "Data center name in a managed Cassandra cluster.",
+      "minLength": 1,
+      "maxLength": 50,
+      "pattern": "^[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*"
+    }
+  },
+  "definitions": {
+    "DefaultErrorResponse": {
+      "description": "An error response from the service.",
+      "properties": {
+        "error": {
+          "description": "The error returned from the service.",
+          "$ref": "cosmos-db.json#/definitions/ErrorResponse"
+        }
+      }
+    },
+    "ListClusters": {
+      "description": "List of managed Cassandra clusters and their properties.",
+      "readOnly": true,
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Container for array of clusters.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ClusterResource"
+          }
+        }
+      }
+    },
+    "ClusterResource": {
+      "x-ms-azure-resource": true,
+      "type": "object",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The unique resource identifier of the ARM resource."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the ARM resource.",
+          "minLength": 1,
+          "maxLength": 100
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The type of Azure resource."
+        },
+        "location": {
+          "type": "string",
+          "description": "The location of the resource group to which the resource belongs."
+        },
+        "tags": {
+          "$ref": "#/definitions/Tags"
+        },
```

Comment:
Let us define ClusterResource to only contain the actual Resource defined by us(no ARM properties). Let us define two new objects: ClusterResourceCreateUpdateProperties that will take in ClusterResource and let us define ClusterResourceGetResults object that will be returned when GET and LIST calls are made. 
Most of the APIs defined in CosmosDB follow this pattern. (You can find eamples for your reference)
These 5 ARM properties are defined in ARMResourceProperties, we can use that object here, instead of redefining them.

Reply:
Same comment for DataCenter resource

Reply:
I was asked not to do this because want to be able to pipe from GET to PUT. I've switched to having them pull their properties from ARMResource and ARMProxyResource in types.json, though.